### PR TITLE
Prettify Dialogs

### PR DESCRIPTION
**Pull request by TheBrokenRail**
Prettify Dialogs, Disable Accessing Mutator UI in Flyout

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly-android/764)

### CODE REVIEW

File: blocklylib-vertical/src/main/java/com/google/blockly/android/ui/vertical/VerticalBlockViewFactory.java [hunk 1]
```diff
@@ -209,7 +209,11 @@ public void onClick(View v) {
                 }
             }
         };
-        return buildIconFieldView(listener, com.google.blockly.android.R.drawable.mutator_icon_24dp);
+        if (!getWorkspaceHelper().getWorkspaceView().getController().getWorkspace().isRootBlock(block.getRootBlock())) {
```
Comment (AnmAtAnm) [inline blocklylib-vertical/src/main/java/com/google/blockly/android/ui/vertical/VerticalBlockViewFactory.java]: Isn't
```!getWorkspaceHelper().getWorkspaceView().getController()```
the same as `this.mController` of the `BlockViewFactory`?
This means the new `getController()` added to `WorkspaceView` is not necessary.

File: blocklylib-core/src/main/java/com/google/blockly/android/ui/WorkspaceView.java [hunk 2]
```diff
@@ -59,6 +59,10 @@ public WorkspaceView(Context context, AttributeSet attrs, int defStyleAttr) {
         mTouchSlop = touchSlop;
     }
 
+    public BlocklyController getController() {
```
Comment (AnmAtAnm) [inline blocklylib-core/src/main/java/com/google/blockly/android/ui/WorkspaceView.java]: Remove, per comment in `VerticalBlockViewFactory`.

File: blocklylib-vertical/src/main/java/com/google/blockly/android/ui/vertical/VerticalBlockViewFactory.java (hunk 1 above)
Comment (TheBrokenRail) [reply blocklylib-vertical/src/main/java/com/google/blockly/android/ui/vertical/VerticalBlockViewFactory.java]: Neither ```BlockViewFactory``` nor ```VerticalBlockViewFactory``` seem to have a ```this.mController``` property.

File: blocklylib-core/src/main/java/com/google/blockly/android/ui/WorkspaceView.java (hunk 2 above)
Comment (TheBrokenRail) [reply blocklylib-core/src/main/java/com/google/blockly/android/ui/WorkspaceView.java]: See my other comment.

File: blocklylib-vertical/src/main/java/com/google/blockly/android/ui/vertical/VerticalBlockViewFactory.java (hunk 1 above)
Comment (AnmAtAnm) [reply blocklylib-vertical/src/main/java/com/google/blockly/android/ui/vertical/VerticalBlockViewFactory.java]: Whoops.  Misread mContext.  But `block.getController()` does exist:
https://github.com/google/blockly-android/blob/develop/blocklylib-core/src/main/java/com/google/blockly/model/Block.java#L197

**Comment by TheBrokenRail** [comment]
@AnmAtAnm

File: blocklylib-vertical/src/main/java/com/google/blockly/android/ui/vertical/VerticalBlockViewFactory.java (hunk 1 above)
Comment (TheBrokenRail) [reply blocklylib-vertical/src/main/java/com/google/blockly/android/ui/vertical/VerticalBlockViewFactory.java]: Done!

**Comment by TheBrokenRail** [comment]
When will this get merged?

**Comment by AnmAtAnm** [comment]
Can you rebase and fix the merge conflict with `blocklylib-core/src/main/res/layout/blockly_variable_name_dialog.xml`?

**Comment by TheBrokenRail** [comment]
Rebased!

**Comment by AnmAtAnm** [comment]
While GitHub is confused, this looks good. GitHub is not recognizing the commits as part of other PRs. Not sure why that is, because the hashes are identical. However, pulling this in via the command line showed the changes are exactly what would be expected:

```
 * branch              prettify-dialogs -> FETCH_HEAD
Auto-merging blocklylib-vertical/src/main/java/com/google/blockly/android/ui/vertical/VerticalBlockViewFactory.java
Merge made by the 'recursive' strategy.
 .../src/main/java/com/google/blockly/android/ui/WorkspaceView.java          |  4 ++++
 .../blockly/android/ui/mutator/ProcedureDefinitionMutatorFragment.java      |  8 ++++---
 blocklylib-core/src/main/res/drawable/ic_add_black_24dp.xml                 |  9 ++++++++
 blocklylib-core/src/main/res/drawable/ic_close_black_24dp.xml               |  9 ++++++++
 blocklylib-core/src/main/res/layout/blockly_variable_name_dialog.xml        |  8 +++++--
 blocklylib-core/src/main/res/layout/if_else_mutator_dialog.xml              | 42 ++++++++++++++++++++++++-------------
 blocklylib-core/src/main/res/layout/name_variable_view.xml                  | 19 +++++++----------
 blocklylib-core/src/main/res/layout/procedure_definition_mutator_add.xml    | 16 +++++++-------
 .../src/main/res/layout/procedure_definition_mutator_argument.xml           | 31 +++++++++++++--------------
 blocklylib-core/src/main/res/layout/procedure_definition_mutator_dialog.xml |  9 ++++----
 .../com/google/blockly/android/ui/vertical/VerticalBlockViewFactory.java    |  6 +++++-
 11 files changed, 102 insertions(+), 59 deletions(-)
```

**Pull request merged**